contracts/CompoundAssetProxy.sol

```diff
@@ -0,0 +1,143 @@
+        // we get the balance before and after minting to calculate shares
+        uint256 beforeBalance = ctoken.balanceOfUnderlying(address(this));
+
+        // Deposit into compound
+        uint256 mintStatus = ctoken.mint(amount);
+        require(mintStatus == 0, "compound mint failed");
+
+        // StoGetre ctoken balance after minting
+        uint256 afterBalance = ctoken.balanceOfUnderlying(address(this));
+        // Calculate ctoken shares minted
+        uint256 shares = afterBalance - beforeBalance;
+        // Return the amount of shares the user has produced and the amount of underlying used for it.
+        return (shares, amount);
+    }
+
+    /// @notice Withdraw the number of shares
+    /// @param _shares The number of shares to withdraw
+    /// @param _destination The address to send the output funds
+    // @param _underlyingPerShare The possibly precomputed underlying per share
+    /// @return Amount of funds freed by doing a withdraw
+    function _withdraw(
+        uint256 _shares,
+        address _destination,
+        uint256
+    ) internal override returns (uint256) {
+        // Since ctoken's redeem function returns sucess codes
+        // we get the balance before and after minting to calculate amount
+        uint256 beforeBalance = token.balanceOf(address(this));
+
+        // Do the withdraw
+        uint256 redeemStatus = ctoken.redeem(_shares);
+        require(redeemStatus == 0, "compound redeem failed");
+
+        // Get underlying balance after withdrawing
+        uint256 afterBalance = token.balanceOf(address(this));
+        // Calculate the amount of funds that were freed
+        uint256 amountReceived = afterBalance - beforeBalance;
+        // Transfer the underlying to the destination
+        // 'token' is an immutable in WrappedPosition
+        token.transfer(_destination, amountReceived);
+
+        // Return the amount of underlying
+        return amountReceived;
+    }
+
+    /// @notice Get the underlying amount of tokens per shares given
+    /// @param _amount The amount of shares you want to know the value of
+    /// @return Value of shares in underlying token
+    function _underlying(uint256 _amount)
+        internal
+        override
+        view
+        returns (uint256)
+    {
+        // Load exchange rate
+        uint256 exchangeRate = ctoken.exchangeRateStored();
+
+        // Multiply _amount by exchange rate & correct for decimals
+        // We assume 8 decimals in the ctoken + 18 point decimal fix = 26
+        return ((_amount * exchangeRate) / (10**(26 - underlyingDecimals)));
+    }
+
+    /// @notice Collect the comp rewards accrued
+    /// @param _destination The address to send the rewards to
+    // TODO: add this back in when I figure out how to mock the owner funcitonality
+    function collectRewards(address _destination) external onlyAuthorized() {
+        // Set up input params for claimComp
+        address[] memory holder = new address[](1);
```

Comment:
Don't need the address array here, we can use the overloaded function `claimComp(address holder, Ctoken[] memory cTokens)`